en_US/durability/durability_introduction.md

```diff
@@ -0,0 +1,133 @@
+
+Durable sessions provide robust durability and high availability by consistently replicating session metadata and MQTT messages across multiple nodes within an EMQX cluster. The configurable [replication factor](./managing-replication.md#replication-factor) determines the number of replicas for each message or session, enabling users to customize the balance between durability and performance to meet their specific requirements.
+
+Advantages of durable client sessions include:
+
+- Sessions can be resumed after EMQX nodes are restarted or stopped.
+- MQTT messages are stored in a shared, replicated, durable storage instead of a memory queue, reducing RAM usage for both online and offline sessions.
+
+However, there are some disadvantages:
+
+- Storing messages on disk results in lower overall system throughput.
+- Durable sessions have higher latency compared to RAM sessions because both writing and reading MQTT messages are performed in batches. While batching improves throughput, it also increases end-to-end latency (the delay before clients see the published messages).
+
+## How Durable Sessions Work
+
+EMQX uses a unique approach to manage message durability, allowing RAM and durable sessions to coexist while minimizing storage costs.
+
+When a durable session subscribes to a topic filter, EMQX marks topics matching the filter as "durable." This ensures that, aside from routing MQTT PUBLISH messages from these topics to RAM sessions, the broker also saves such messages to the durable storage.
+
+Each durable MQTT message is stored exactly once on each replica, regardless of the number of subscribing durable sessions or their connection status. This efficient fan-out minimizes disk writes.
+
+### Durable Storage Architecture
+
+EMQX's durable storage is organized into a hierarchical structure comprising storages, shards, generations, and streams.
+
+![Diagram of EMQX durable storage sharding](./assets/emqx_ds_sharding.png)
+
+#### Storage
+
+Storage encapsulates all data of a certain type, such as MQTT messages or MQTT sessions.
+
+#### Shard
+
+Messages are segregated by client and stored in shards based on the publisher's client ID. The number of shards is determined by [n_shards](./managing-replication.md#number-of-shards) configuration parameter during the initial startup of EMQX. A shard is also a unit of replication. Each shard is consistently replicated the number of times specified by `durable_storage.messages.replication_factor` across different nodes, ensuring identical message sets in each replica.
+
+#### Generation
+
+Messages within a shard are segmented into generations corresponding to specific time frames. New messages are written to the current generation, while previous generations are read-only. Old generations are deleted based on the `durable_sessions.message_retention_period` parameter.
```

Comment:
@Meggielqk I think an important point got lost here, that _MQTT messages_ are cleaned up be deleting the generations containing them. Users don't care about generations, they care about the messages, so this part should be emphasized.

Reply:
updated